scripts/pylib/twister/twisterlib.py

```diff
@@ -2025,7 +2025,7 @@ def scan_path(self, path):
             except ValueError as e:
                 logger.error("%s: can't find: %s" % (filename, e))
 
-        for filename in glob.glob(os.path.join(path, "*.c")):
+        for filename in glob.glob(os.path.join(src_dir_path, "*.c")):
```

Comment:
According to [here](https://docs.python.org/3/library/glob.html#glob.glob).
We are not setting the `recursive=True` and it is `False` by default for the `glob()` call. And we are not using `**` pattern. So it will only search for .c file immediate in the "path" folder where testcase.yaml lives, which normally has no .c file.

So with current implementation, we only get .c files from src/ folder.

Reply:
right, need be "**/*.c", then question is why we do not meet any issue before, if suppose this code does not work.

Reply:
I think because we are lucky to have all source code immediately in src as of now...

Reply:
And similar issue exists here: https://github.com/zephyrproject-rtos/zephyr/blob/ce84c782dcd3cee6fad6ac7ec7642accdaa42d1f/scripts/pylib/twister/twisterlib.py#L3311

`for file in glob.glob(os.path.join(board_root, "*", "*", "*.yaml")):`